core/src/test/java/org/elasticsearch/search/internal/InternalSearchHitTests.java

```diff
@@ -94,4 +215,45 @@ public void testHasSource() {
         searchHit.sourceRef(new BytesArray("{}"));
         assertTrue(searchHit.hasSource());
     }
+
+    public void testParseShardId() {
```

Comment:
It is a bit odd that we are testing here the `ShardId#fromString` method. In fact I am not even sure this method is generic enough to belong in `ShardId`...